server/registration/services/RegistrationService.js

```diff
@@ -115,6 +115,112 @@ module.exports = {
         }
       );
     }
+  },
+
+  /**
+   * Confirm registration
+   *
+   * @param {object} request - The request that was sent from the controller
+   */
+  confirmRegistration: async function confirmRegistration(request) {
+
+    let registration = null;
+
+    try {
+      registration = await Verification.findOne(
+        {
+          msisdn: request.msisdn
+        }
+      );
+
+      // check if there is an existing registration code
+      if (appUtil.isNullOrUndefined(registration)) {
+        return internalEventEmitter.emit(
+          registrationChannels.Internal.ConfirmRegistrationCompletedEvent,
+          {
+            statusCode: 400,
+            body: '' // send empty?
+          }
+        );
+      }
+
+      // registration already confirmed previously
+      if (registration.status == true) {
+
+        logging.logAction(
+          logging.logLevels.INFO,
+          'Registration for ' + request.msisdn + ' already confirmed previously'
```

Comment:
again just FYI , you can use string interpolation here, so your message can look like this:

```
`Registration for [${request.msisdn}] already confirmed previously`
```